tests/unittests/test_client.py

```diff
@@ -0,0 +1,211 @@
+import pytest
+from unittest.mock import patch
+from datetime import datetime
+import requests
+from requests.exceptions import ConnectionError, Timeout, ChunkedEncodingError
+
+from tap_ms_graph.client import Client, raise_for_error
+from tap_ms_graph.exceptions import (
+    ERROR_CODE_EXCEPTION_MAPPING,
+    MsGraphError,
+    MsGraphUnauthorizedError,
+    MsGraphBadRequestError,
+    MsGraphRateLimitError,
+    MsGraphInternalServerError,
+)
+
+
+class MockResponse:
+    def __init__(self, status_code, json_data=None, raise_error=False, headers=None, text=None):
+        self.status_code = status_code
+        self._json_data = json_data or {}
+        self.raise_error = raise_error
+        self.headers = headers or {}
+        self.text = text or ""
+
+    def raise_for_status(self):
+        if self.raise_error:
+            raise requests.HTTPError("Sample message")
+        return self.status_code
+
+    def json(self):
+        return self._json_data
+
+
+def get_response(status_code, json_data=None, headers=None, raise_error=False):
+    return MockResponse(status_code, json_data, raise_error, headers)
+
+
+@pytest.mark.parametrize("status_code", [200, 201, 204])
+def test_raise_for_error_success(status_code):
+    response = get_response(status_code, {"message": "OK"})
+    raise_for_error(response)  # Should not raise
+
+
+@pytest.mark.parametrize(
+    "response_data, expected_exception, expected_msg_part",
+    [
+        ({"code": "BadRequest", "details": "Invalid input"}, MsGraphBadRequestError, "HTTP-error-code: 400"),
+        ({"message": "Unauthorized access"}, MsGraphUnauthorizedError, "Unauthorized access"),
+        (None, MsGraphInternalServerError, "HTTP-error-code: 500"),
+        ({"message": "Unknown error received from the API."}, MsGraphError, "Unknown error"),
+    ],
```

Comment:
To improve readability.

```suggestion
    [
        ({"code": "BadRequest", "details": "Invalid input"},
         MsGraphBadRequestError,
         "HTTP-error-code: 400"),
        ({"message": "Unauthorized access"},
         MsGraphUnauthorizedError,
         "Unauthorized access"),
        (None,
         MsGraphInternalServerError,
         "HTTP-error-code: 500"),
        ({"message": "Unknown error received from the API."},
         MsGraphError,
         "Unknown error"),
    ],
```

Reply:
addressed